### PR TITLE
Jd/agc model

### PR DESCRIPTION
**Pull request by jd-lara**
This is WIP, the review is mostly for very bad errors.

### CODE REVIEW

File: src/services_models/services_constructor.jl [hunk 1]
```diff
@@ -72,3 +72,32 @@ function construct_service!(
...
 end
+
+function construct_service!(
+    psi_container::PSIContainer,
+    services::IS.FlattenIteratorWrapper{PSY.AGC},
+    services_mapping::PSY.ServiceContributingDevicesMapping,
+    ::ServiceModel{PSY.AGC, T},
+    devices_template::Dict{Symbol, DeviceModel},
+) where {T <: AbstractAGCFormulation}
+    #Note: Frequency is a global variable
```
Comment (daniel-thom) [inline src/services_models/services_constructor.jl:83]: I don't follow the comment.

File: src/services_models/services_constructor.jl [hunk 2]
```diff
@@ -72,3 +72,32 @@ function construct_service!(
...
+    steady_state_frequency_variables!(psi_container)
+
+    for device_model in devices_template
+    end
```
Comment (daniel-thom) [inline src/services_models/services_constructor.jl:87]: Is this supposed to do something?

File: src/services_models/services_constructor.jl [hunk 3]
```diff
@@ -72,3 +72,32 @@ function construct_service!(
...
+            services_mapping[(
+                type = typeof(service),
+                name = PSY.get_name(service),
+            )].contributing_devices
+        #Variables
+        regulation_service_variables!(psi_container, service, contributing_devices)
+        # Constraints
+        smooth_ace_pid!(psi_container, service)
+        #steady_state_frequency_variables!(psi_container, service, model, )
+        #participation_assignment!(psi_container, service, model, contributing_devices)
```
Comment (daniel-thom) [inline src/services_models/services_constructor.jl:100]: stale

File: src/services_models/services_constructor.jl (hunk 1 above)
Comment (jd-lara) [reply src/services_models/services_constructor.jl:83]: The Frequency of the system is the same all over not just for an area.

File: src/services_models/services_constructor.jl (hunk 2 above)
Comment (jd-lara) [reply src/services_models/services_constructor.jl:87]: a check goes here. Not implemented yet

File: src/services_models/services_constructor.jl (hunk 3 above)
Comment (jd-lara) [reply src/services_models/services_constructor.jl:100]: It is a broken code.

**Pull request merged**